aten/src/ATen/NumericUtils.h

```diff
@@ -51,5 +51,56 @@ inline C10_HOST_DEVICE bool _isnan(at::BFloat16 val) {
   return at::_isnan(float(val));
 }
 
+template <typename T,
+          typename std::enable_if<std::is_same<T, double>::value, int>::type = 0>
```

Comment:
Do you really need a template enable if here? As far as I can tell, you are only using these functions for `float` and `double` overloads. Is the idea to plan for a hypothetical complex overload later? (cc @zasdfgbnm )

Reply:
I've run into the issue in the past when this was needed to work around ambiguities when same type was aliased across namespaces with `using foo = bar::foo;` or somesuch.

Reply:
addressed